p2p/config/config.go

```diff
@@ -0,0 +1,86 @@
+package config
+
+import (
+	"fmt"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/pokt-network/pocket/p2p/providers"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"go.uber.org/multierr"
+)
+
+// baseConfig implements `RouterConfig` using the given libp2p host and current
+// height and peerstore providers. Intended for internal use by other `RouterConfig`
+// implementations with common config parameters.
+//
+// NB: intentionally *not* embedding `baseConfig` to improve readability of usages
```

Comment:
I don't fully understand this. So when would you embed a struct inside of another one?

To me this seems like a worthwhile usecase

Reply:
I'm suggesting never. It doesn't look/feel good from the usage side:

```golang
	bgCfg := BackgroundConfig{
		BaseConfig: BaseConfig{
			Host:                  nil,
			Addr:                  nil,
			CurrentHeightProvider: nil,
			PeerstoreProvider:     nil,
		},
	}

	rtCfg := RainTreeConfig{
		BaseConfig: BaseConfig{
			Host:                  nil,
			Addr:                  nil,
			CurrentHeightProvider: nil,
			PeerstoreProvider:     nil,
		},
		MaxNonces: 100,
	}
```

Seeing this :point_up_2: makes me want to write a `NewRainTreeConfig()` function which would be a bit ridiculous IMHO given that that the purpose of the config originally was to simplify the `NewRainTreeRouter()` function signature.

Additionally, I expect significant simplification may be around the corner for the router interface. I'm anticipating being able to eliminate the need for the background router to hold a reference to either provider (maybe even the pokt addr as  well). This would cause further divergence between `BackgroundConfig` and `RainTreeConfig`, likely making `baseConfig` another temporary step.

Reply:
I feel strongly about not taking embedding seriously as an option because of the impact on readability, given the context. An alternative to defining `baseConfig`  would be to duplicate the `IsValid()` implementation across both `RainTreeConfig` and `BackgroundConfig`; however, I see the way it's currently written as a simplification of that.

Reply:
![giphy](https://user-images.githubusercontent.com/1892194/236554649-91bf8ad4-a531-4160-8b16-a6d791fa47cc.gif)